src/visitor.rs

```diff
@@ -120,11 +120,14 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
             ast::StmtKind::Local(..) | ast::StmtKind::Expr(..) | ast::StmtKind::Semi(..) => {
                 let attrs = get_attrs_from_stmt(stmt.as_ast_node());
                 if contains_skip(attrs) {
+                    let span = stmt.span();
                     self.push_skipped_with_span(
                         attrs,
-                        stmt.span(),
+                        span,
                         get_span_without_attrs(stmt.as_ast_node()),
                     );
+                    let context = self.get_context();
+                    add_skipped_range(&context, span);
```

Comment:
`push_skipped_with_span` is super tricky. Even though we skip formatting any statement/item/expression with `rustfmt::skip` attribute, we do **not** want to skip formatting the attribute itself. E.g.,

```rust
#[rustfmt::skip]  

let x = 3  ;
```

in the above example, even though we do not modify `let x = 3  ;`, we do want to trim trailing spaces after `#[rustfmt::skip]` itself and any empty lines between the attribute and the let statement (though I don't think we do this correctly in every case).

Reply:
The takeaway here is that we do not want to call `add_skipped_range` with a span which includes `#[rustfmt::skip]`.